daml-lf/language/src/main/scala/com/digitalasset/daml/lf/language/Ast.scala

```diff
@@ -578,8 +578,8 @@ object Ast {
     def apply(typ: Type, noPartyLiterals: Boolean, body: E, isTest: Boolean): GenDValue[E] =
       new GenDValue(typ, noPartyLiterals, body, isTest)
 
-    def unapply(arg: GenDValue[E]): Option[(Type, Boolean, E, Boolean)] =
-      GenDValue.unapply(arg)
+    def unapply(arg: GenDValue[E]): Some[(Type, Boolean, E, Boolean)] =
```

Comment:
While `Option` results in exhaustive warnings `Some` does not. However, the autogenerated `unapply` on a case class returns `Option`. We could either use the underlying `unapply` and just `.get` it or inline it here. Given that these don’t change too often, I’ve chosen the latter but I don’t feel strongly about this.